service/src/main/kotlin/fi/espoo/evaka/application/ApplicationControllerV2.kt

```diff
@@ -262,7 +262,7 @@ class ApplicationControllerV2(
             ResponseEntity.ok(
                 ApplicationResponse(
                     application = application,
-                    decisions = decisions.map { it.copy(documentUri = null) },
+                    decisions = decisions,
```

Comment:
If I recall correctly, the point of this was to not expose s3 uri outside. Preferrably the field would not even exist in the response type, but having an almost identical type for it and mapping between them is also annoying, since in kotlin we dont have something like `({documentUri, ...rest}) => ({...rest})`

Reply:
inheritance might be one way to solve this but then we can't use `data class`

Reply:
I stumbled across this solution in `DecisionController` which actually feels like a nice middle ground between the current change and implementing a whole separate interface/sealed class just for `Decision`:

```kotlin
fun withPublicDocumentUri(decisions: List<Decision>) = decisions.map { decision ->
    decision.copy(documentUri = "/api/internal/decisions2/${decision.id}/download") // do not expose direct s3 uri
}
```

I'd maybe move that to the data class itself, so the full solution would be like:

```kotlin
data class Decision(...) {
    // other methods...
    fun withPublicDocumentUris() = this.copy(
            documentUri = if (this.documentUri != null) "/api/internal/decisions2/${this.id}/download" else null,
            otherGuardianDocumentUri = if (this.otherGuardianDocumentUri != null) "/api/internal/decisions2/${this.id}/download" else null
    ) // Do not expose direct S3 URIs
}

// Usage:
val decisions = getDecisionsByApplication(tx.handle, applicationId, acl.getAuthorizedUnits(user))
        .map { it.withPublicDocumentUris() }
```

-> no need for a separate mapping & solves the exposed S3 URL issue (though that's IMO more of a smell than an actual issue, as the bucket still has be properly access controlled).

Reply:
:thinking: Also, looks like the original S3 URIs are currently returned from quite a few endpoints (obviously as they're part of the data model and it requires explicitly obscuring them).

Reply:
Security-wise I don't feel exposing the URIs is a major issue so my suggestion is: go forward with this as-is (+E2E test fixes) and take tech debt for overhauling the decision data model so that it doesn't have the full URIs at all (just filenames/IDs and anything further is just download-time configuration). This current is issue is such an edge case, I don't think I should use too much time on this right now. WDYT?

Reply:
Sounds good to me 👍